core/src/banking_stage.rs

```diff
@@ -2108,4 +2098,177 @@ mod tests {
+        Blockstore::destroy(&ledger_path).unwrap();
+    }
+
+    #[test]
+    fn test_consume_buffered_packets_interrupted() {
+        let ledger_path = get_tmp_ledger_path!();
+        {
+            let (transactions, bank, poh_recorder, _entry_receiver) =
+                setup_conflicting_transactions(&ledger_path);
+            let num_conflicting_transactions = transactions.len();
+            let packets = to_packets_chunked(&transactions, 1);
+            assert_eq!(packets.len(), num_conflicting_transactions);
+            for single_packets in &packets {
+                assert_eq!(single_packets.packets.len(), 1);
+            }
+            let mut buffered_packets: UnprocessedPackets = packets
+                .clone()
+                .into_iter()
+                .map(|single_packets| (single_packets, vec![0]))
+                .collect();
+
+            let (continue_sender, continue_receiver) = unbounded();
+            let (finished_packet_sender, finished_packet_receiver) = unbounded();
+
+            let test_fn = Some(move || {
+                finished_packet_sender.send(1).unwrap();
+                continue_receiver.recv().unwrap();
+            });
+            // When the poh recorder has a bank, should process all non conflicting buffered packets.
+            // Processes one packet per iteration of the loop
+            let interrupted_iteration = 1;
+            poh_recorder.lock().unwrap().set_bank(&bank);
+            let poh_recorder_ = poh_recorder.clone();
+            let (gossip_vote_sender, _gossip_vote_receiver) = unbounded();
+            // Start up thread to process the banks
+            let t_consume = Builder::new()
+                .name("consume-buffered-packets".to_string())
+                .spawn(move || {
+                    BankingStage::consume_buffered_packets(
+                        &Pubkey::default(),
+                        &poh_recorder_,
+                        &mut buffered_packets,
+                        None,
+                        &gossip_vote_sender,
+                        test_fn,
+                    );
+
+                    // Check everything is correct. All indexes after `interrupted_iteration`
+                    // should still be unprocessed
+                    assert_eq!(
+                        buffered_packets.len(),
+                        num_conflicting_transactions - interrupted_iteration - 1
+                    );
+                    for ((remaining_unprocessed_packet, _), original_packet) in buffered_packets
+                        .iter()
+                        .zip(&packets[interrupted_iteration + 1..])
+                    {
+                        assert_eq!(
+                            remaining_unprocessed_packet.packets[0],
+                            original_packet.packets[0]
+                        );
+                    }
+                })
+                .unwrap();
+
+            for i in 0..=interrupted_iteration {
```

Comment:
nits: `for interrupt_now in [false, true] {`

Reply:
hmmm `interrupted_iteration` isn't necessarily equal to `1`, so this is not always equivalent right?

Reply:
odd? it's not `mut`:

https://github.com/solana-labs/solana/pull/15210/files#diff-bb8464c2f02c8863fb60e6e29f023f3e6da93a93fbb9199a70a44c9c1b41c7f3R2332
> `let interrupted_iteration = 1;`

Maybe, some debug leftover?

Reply:
oh I meant I prefer it remains just a placeholder value, i.e., we could make it `0, 2` in the future